authenticity-collector/src/sludata/sludata.service.ts

```diff
@@ -0,0 +1,37 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { ApiVersion, ChannelClient, ChannelData, ChannelInfo, ClientConfig } from 'iota-is-sdk/lib';
+import { firstValueFrom } from 'rxjs';
+import { SluDataDto } from './model/SluDataDto';
+import * as fs from 'fs'
+import { AES } from 'crypto-js';
+
+@Injectable()
+export class SludataService {
+	constructor(
+		private configService: ConfigService,
+		private httpService: HttpService,
+	) { }
+
+	async writeData(data: SluDataDto): Promise<ChannelData> {
+		const mpowerUrl = this.configService.get<string>('MPOWER_CONNECTOR_URL');
+		await firstValueFrom(this.httpService.post(mpowerUrl, data));
+
+		const collectorIdPath = this.configService.get<string>('COLLECTOR_ID_PATH');
+		const collectorJson = fs.readFileSync(collectorIdPath, 'utf-8');
+		const collector = JSON.parse(collectorJson);
+		const clientConfig: ClientConfig = {
+			apiKey: this.configService.get('IS_API_KEY'),
+			baseUrl: this.configService.get('IS_API_URL'),
+			apiVersion: ApiVersion.v01
+		};
+		const channelClient = new ChannelClient(clientConfig);
+		await channelClient.authenticate(collector.doc.id, collector.key.secret);
+		const channelInfo: ChannelInfo[] = await channelClient.search({ authorId: collector.doc.id });
+
+		const hashedData = AES.encrypt(JSON.stringify(data.payload), collector.doc.id).toString();
```

Comment:
seems like you encrypt the data but you should hash it. see here for instance:
https://github.com/iotaledger/integration-services/blob/master/api/src/utils/encryption/index.ts#L13